src/sfizz/modulations/sources/Controller.cpp

```diff
@@ -85,27 +87,105 @@ void ControllerSource::init(const ModKey& sourceKey, NumericId<Voice> voiceId, u
 
 void ControllerSource::generate(const ModKey& sourceKey, NumericId<Voice> voiceId, absl::Span<float> buffer)
 {
-    (void)voiceId;
-
     const ModKey::Parameters p = sourceKey.parameters();
     const Resources& res = *impl_->res_;
     const Curve& curve = res.curves.getCurve(p.curve);
     const MidiState& ms = res.midiState;
-    const EventVector& events = ms.getCCEvents(p.cc);
+    bool canShortcut = false;
 
     auto transformValue = [p, &curve](float x) {
-        return curve.evalNormalized(x);
+        // Otherwise it clamps the bipolar values
+        if (p.curve != 0)
```

Comment:
This is not entirely satisfying but it's needed because the default curve (and all curves) "clamp" their inputs.

Reply:
Should bipolar curves not be evaluated centered at 0.5 ?

Reply:
After some tests, it seems the curve is ignored for at least cc128 (bend) in Sforzando:
`<region> sample=*sine pitch_cc128=200 pitch_curvecc128=3` does not invert the polarity. In fact setting any predefined curve does nothing.

Reply:
`stepcc` does work though.

Reply:
Actually curve seems to be ignored in sforzando for extended ccs. I matched this behavior.